ITE/2/README.adoc

```diff
@@ -0,0 +1,532 @@
+compromise. Security attacks are listed as columns in increasing order
+of severity. The result of combining two or more functionaries / roles
+is that "`Yes`" overrides "`No`" or "`N/A`".
+
+[[infrastructure-requirements]]
+== Infrastructure requirements
+
+The changes discussed here require adding _at least_ the following
+infrastructure and resources:
+
+* *An offline key ceremony to generate and rotate all private and public
+keys for TUF and in-toto.* Operations security (OPSEC) is extremely
+important here into ensuring that offline private keys are not leaked or
+exported by accident. Enough backups should also be made to ensure that
+the keys for the entire system can be rotated when necessary, such as
+removing or adding developer keys, which is an additional cost.
+* *Provisioning developers with signing keys and developer tools with which they
+can sign their own code.* Costs include ease-of-use, and protecting keys
+from being exfiltrated, possibly using HSMs.
+* *Instrumenting every step in the CI/CD pipeline to generate signed
+in-toto metadata.* Costs include protecting keys from being exfiltrated,
+possibly using HSMs.
+* *Adding or instrumenting a last step in the CI/CD pipeline to bundle
+and sign all in-toto metadata and packages using TUF.* There should be a
+scheduled task that automatically runs at fixed intervals in order to
+renew signatures and prevent expiration of TUF metadata.
+* *Instrumenting package managers to download and verify metadata using
+TUF and in-toto before installing new code.*
+* *Time to test the entire pipeline end-to-end between developers and
+end-users, and providing support for unforeseen issues.* Costs include
+updating the package managers in order to fix these issues.
+
+[[backwards-compatibility]]
+== Backwards compatibility
+
+The changes discussed here require considering _at least_ the following
+issues of backwards compatibility:
+
+* *If previous package managers already use in-toto but not TUF, then
+the filesystem structure for in-toto metadata and packages should be
+preserved for package managers that have not yet been updated to use
+TUF.* These package managers should be updated to newer versions which
+use both TUF and in-toto.
+* *When rotating TUF root keys, all versions of the root TUF metadata
+from version 1 to N, where N is the latest version, should be preserved
+so that package managers can update from one version to the next.* This
+assumes that any version N+1 is signed by at least a threshold of root
+keys listed in version N.
+* *Package managers should use
+https://github.com/theupdateframework/taps/blob/master/tap6.md[TAP **6]*
+to determine how to respond when a TUF repository has significantly
+updated its specification version.**
+* *Neither a TUF nor in-toto implementation should make
+backward-incompatible changes to how signatures are generated, so that
+previous package managers are able to continue to install new packages.*
```

Comment:
This may be tricky as support for crypto agility is added.  In the future, there is hope that significant future-proofing properties can be maintained.